packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -145,9 +145,13 @@ class Doctor {
     for (ValidatorTask validatorTask in startValidatorTasks()) {
       final DoctorValidator validator = validatorTask.validator;
       final Status status = new Status.withSpinner();
-      await (validatorTask.result).then<void>((_) {
-        status.stop();
-      }).whenComplete(status.cancel);
+      try {
+        await validatorTask.result;
+      } catch (exception) {
+        status.cancel();
+        rethrow;
+      }
+      status.stop();
```

Comment:
I think it reads a little clearer is you move this into the try block

Reply:
if i do that and the stop itself fails, it would then try to cancel, which would likely also fail, which would then mask the original failure. I agree that it reads better. I don't have a strong opinion; which way do you want it?

Reply:
Hm, good point.  Let's leave it as-is.